cg_lims/EPPs/files/placement_map/make_96well_placement_map.py

```diff
@@ -0,0 +1,170 @@
+    return "".join(html)
+
+
+def make_html(placement_map: dict, process: Process, original_well: str) -> str:
+    """Creating the html."""
+
+    html = []
+    html.append("<html>")
+    header_info = PlacementMapHeader(process_type=process.type.name, date=date.today().isoformat())
+    html.append(PLACEMENT_MAP_HEADER.format(**header_info.dict()))
+    container_type = "Original" if original_well else "Source"
+    for container, container_info in placement_map.items():
+        plate_info = PlateInfo(
+            container_name=container.name,
+            container_type=container.type.name,
+            container_id=container.id,
+        )
+        html.append(""""<table class="group-contents"><br><br><thead>""")
+        html.append(PLATE_HEADER_SECTION.format(**plate_info.dict()))
+        html.append(TABLE_HEADERS.format(container_type=container_type))
+        html.append("</thead>")
+        html.append("<tbody>")
+        for dest_well, well_data in container_info.items():
+            html.append(TABLE_ROWS.format(**well_data.dict()))
+        html.append("</tbody></table><br><br>")
+
+        ## VISUAL Platemap
+
+        html.append(VISUAL_PLACEMENT_MAP_HEADER)
+        html.append("<tbody>")
+
+        coulmns = range(1, 14)
+        rows = ["A", "B", "C", "D", "E", "F", "G", "H"]
+        for rowname in rows:
+            html.append(
+                f"""
+                <tr style="height: 12%;">
+                <td class="bold-column row-name">{rowname}</td>"""
+            )
+            for column in coulmns:
+                well_location = f"{rowname}:{column}"
+                if well_location in container_info:
+                    # This only happens if there is an artifact in the well
+                    # This assumes that all artifacts have the required UDFs
+                    well_info = container_info[well_location]
+                    html.append(VISUAL_PLACEMENT_MAP_WELL.format(**well_info.dict()))
+                else:
+                    # For wells that are empty:
+                    html.append('<td class="well" style="">&nbsp;</td>')
+            html.append("</tr>")
+        html.append("</body></table>")
+    html.append("</html>")
+    html = "".join(html)
+    return html
+
+
+@click.command()
+@options.file_placeholder(help="File placeholder name.")
+@options.sample_udfs(help="Sample UDFs to show in the placement map.")
+@options.original_well()
+@click.pass_context
+def placement_map(ctx, file: str, sample_udfs: List[str], original_well: str):
+    """Create a 96 well placement map."""
+
+    LOG.info(f"Running {ctx.command_path} with params: {ctx.params}")
+    process = ctx.obj["process"]
+    lims = ctx.obj["lims"]
+
+    try:
+        placement_map_data = get_placement_map_data(lims, process, sample_udfs, original_well)
+        html = make_html(placement_map_data, process, original_well)
+        file = open(f"{file}.html", "w")
+        file.write(html)
+        file.close()
```

Comment:
```suggestion
        file = open(f"{file}.html", "w")
        file.write(html)
        file.close()
```
        with open(f"{file}.html") as filehandler:
            filehandler.write(html)